### PR TITLE
ARC-1243: Handle Internationalisation based on some property - Add deploy IT action

### PR DESCRIPTION
**Pull request by grazianomarallo-oviva**
-  Add deploy IT action

### CODE REVIEW

File: .github/workflows/deploy-it.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        with:
+          token_format: access_token
+          workload_identity_provider: projects/155692196363/locations/global/workloadIdentityPools/github-actions-sa/providers/github-actions-sa
+          service_account: gh-wi-main-registry-writer-gh@github-actions-sa.iam.gserviceaccount.com
+      - name: Login to Google Artifact Registry
+        uses: docker/login-action@v2
+        with:
+          registry: europe-docker.pkg.dev
+          username: oauth2accesstoken
+          password: ${{ steps.auth.outputs.access_token }}
```
Comment (moritzschmitz-oviva) [inline .github/workflows/deploy-it.yml:45]:
```suggestion
      - name: Login to Google Artifact Registry
        uses: docker/login-action@v3
        with:
          registry: ghcr.io
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/deploy-it.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          cache: 'maven'
+      - name: Authenticate to Google Cloud
+        id: auth
+        uses: google-github-actions/auth@v2
+        with:
+          token_format: access_token
+          workload_identity_provider: projects/155692196363/locations/global/workloadIdentityPools/github-actions-sa/providers/github-actions-sa
+          service_account: gh-wi-main-registry-writer-gh@github-actions-sa.iam.gserviceaccount.com
```
Comment (moritzschmitz-oviva) [inline .github/workflows/deploy-it.yml:39]:
```suggestion
```

File: .github/workflows/deploy-it.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    env:
+      NAMESPACE: 'ehealthid-relying-party'
+      DOCKER_REPO: 'europe-docker.pkg.dev/oviva-pkg/ovi/'
+      IMAGE_NAME_BACKEND: 'ehealthid-relying-party'
```
Comment (moritzschmitz-oviva) [inline .github/workflows/deploy-it.yml:17]:
```suggestion
```

**Review by demianjaeger-oviva — CHANGES_REQUESTED** [review]
Hi all

Let's first discuss this together in our catchup. I think we should not directly connect to our clusters from public repos. One wrong log output exposes then credentials to the public.

- [ ] let's discuss this first together

File: .github/workflows/deploy-it.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    env:
+      NAMESPACE: 'ehealthid-relying-party'
+      DOCKER_REPO: 'europe-docker.pkg.dev/oviva-pkg/ovi/'
```
Comment (moritzschmitz-oviva) [inline .github/workflows/deploy-it.yml:16]: These values are all set in the Makefile already.
```suggestion
```

File: .github/workflows/deploy-it.yml (hunk 3 above)
Comment (moritzschmitz-oviva) [inline .github/workflows/deploy-it.yml:17]: Namespace is not used at all inside the file. Repo needs to be changed to the GitHub Container Registry.
```suggestion
      DOCKER_REPO: 'ghcr.io/oviva-ag/'
      IMAGE_NAME_BACKEND: 'ehealthid-relying-party'
```

**Review by moritzschmitz-oviva — COMMENTED** [review]
Some adjustments required here. Also agree with Demian here.

File: .github/workflows/deploy-it.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          author_email: github-actions@oviva.internal
+          branch: main
+          message: 'CI: ehealthid-relying-party: Update IT docker image'
+          pull_strategy: '--no-rebase'
+          push: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_CI_PAT }}
+    permissions:
+      contents: read
+      id-token: write
```
Comment (moritzschmitz-oviva) [inline .github/workflows/deploy-it.yml:75]:
```suggestion
    permissions:
      contents: read
      packages: write
```

**Pull request closed without merging**